rbi/core/dir.rbi

```diff
@@ -567,18 +567,10 @@ class Dir < Object
   # Equivalent to calling `Dir.glob([string,...], 0)`.
   sig do
     params(
-        pattern: T.any(String, T::Array[String]),
-        flags: Integer,
+        pattern: String,
+        base: T.nilable(String)
     )
     .returns(T::Array[String])
   end
-  sig do
-    params(
-        pattern: T.any(String, T::Array[String]),
-        flags: Integer,
-        blk: T.proc.params(arg0: String).returns(BasicObject),
-    )
-    .returns(NilClass)
-  end
-  def self.[](pattern, flags=T.unsafe(nil), &blk); end
+  def self.[](*pattern, base: nil); end
```

Comment:
I think we should still keep the optional `&blk` argument in this signature since:
```ruby
Dir["/*"] do |match|
  puts match
end
```
works.

Reply:
Thanks, I'll add back a signature with `blk`. Should the block option be added to the `Dir#[]` documentation?

Reply:
The documentation on top the methods here are usually just copied over from the Ruby documentation without changes. Not sure if the Sorbet team makes any changes to them to improve.

It might be best to improve Ruby documentation regarding this in the long run, which would ultimately improve it here too.

Reply:
We don't usually make any changes to the ruby docs 👍